File: TaskVoteUI/src/app/model/taskVoteDTO.ts
import { Vote } from "./vote";

/*
interface Serializable<T> {
    deserialize( input: Object ): T;
}*/

export class TaskVoteDTO {
    id: string;
    vote: Vote;

    getId(): string {
        return this.id;
    }
    setId(id: string) : void {
       this.id = id; 
    }

    getVote() : Vote {
        return this.vote;
    }
    setVote(vote: Vote) : void {
        this.vote = vote;
    }

    /*
    deserialize( json: any ) {
        this.id = json.id;
        this.vote = json.vote;
        return this;
    }*/
}

File: TaskVoteApp/src/main/java/com/example/demo/dto/TaskVoteDTO.java
package com.example.demo.dto;

import com.example.demo.model.Vote;

public class TaskVoteDTO {

	private String id;
	private Vote vote;
	
	public String getId() {
		return id;
	}
	public void setId(String id) {
		this.id = id;
	}
	public Vote getVote() {
		return vote;
	}
	public void setVote(Vote vote) {
		this.vote = vote;
	}
}
